Found the shard-2 failure for you — it is not in the code you wrote, and the fix is three words long.

`load_parked_sessions` in `src/bernstein/core/orchestration/supervisor_aggregator.py:220` reads the marker at `.sdd/runtime/spawn_supervisor/parked.json` and then **unions** in every `respawn_exhausted` record it finds under `.sdd/runtime/failures/`. That union was harmless only because nothing wrote those records yet. Your PR writes them, and a record there is a historical fact ("this session once exhausted its respawns") rather than a statement that it is parked *now*. `clear_parked` and `resume` rewrite the marker and neither rewrites history, so a cleared park comes straight back:

```
assert result.session_ids == frozenset()
E   AssertionError: assert frozenset({'batch:T-6'}) == frozenset()
```

The marker should win whenever it answers:

```diff
-    # Lifecycle events fallback: scan the failures/ records.
+    # Lifecycle events fallback: scan the failures/ records -- but only when
+    # the marker file did not answer. A record under failures/ states a
+    # historical fact, not that the session is parked now; unioning it into a
+    # marker that exists resurrects every id the marker deliberately cleared.
     failures_dir = workdir / ".sdd" / "runtime" / "failures"
-    if failures_dir.exists():
+    if not available and failures_dir.exists():
```

I verified this locally: `tests/unit/agents/test_parked_session_producer.py` + `tests/unit/test_supervisor_aggregator.py` go 2 failed → 20 passed, ruff clean. Worth adding a test alongside it so the union cannot come back — seed a marker with `{"session_ids": []}`, drop one `respawn_exhausted` record next to it, and assert `available` is true while `session_ids` is empty.

I could not push to your fork branch directly. If you would rather I carried it, enable "allow edits from maintainers" on the PR and I will land it; otherwise take the patch above and it is yours. Your `_write_failure_record` itself is right — atomic write via a `.tmp` and `os.replace`, and an `OSError` handler that actually reports.

--- a/src/bernstein/core/orchestration/supervisor_aggregator.py
+++ b/src/bernstein/core/orchestration/supervisor_aggregator.py
@@ -217,9 +217,15 @@ def load_parked_sessions(workdir: Path) -> ParkedSessions:
             if isinstance(ids, list):
                 ids_list = cast(list[Any], ids)
                 parked.update(str(i) for i in ids_list if isinstance(i, str))
-    # Lifecycle events fallback: scan the failures/ records.
+    # Lifecycle events fallback: scan the failures/ records. A fallback, not
+    # a second source to union in: a `respawn_exhausted` record states that a
+    # session was once exhausted, which is history, not present state. Unioned
+    # unconditionally it resurrects every id that `clear_parked` or a resume
+    # has since removed, because the marker it was cleared from is still there
+    # and still says nothing about it. Read it only when the marker gave no
+    # answer at all -- absent, or unreadable.
     failures_dir = workdir / ".sdd" / "runtime" / "failures"
-    if failures_dir.exists():
+    if not available and failures_dir.exists():
         with suppress(OSError):
             for entry in sorted(failures_dir.glob("*.json")):
                 with suppress(OSError, json.JSONDecodeError):

--- a/tests/unit/agents/test_parked_session_producer.py
+++ b/tests/unit/agents/test_parked_session_producer.py
@@ -220,3 +220,48 @@ def test_a_corrupt_store_does_not_stop_a_park_being_recorded(tmp_path: Path) -> 
     SpawnSupervisor(workdir=tmp_path).park("batch:T-10")
 
     assert load_parked_sessions(tmp_path).session_ids == frozenset({"batch:T-10"})
+
+
+def test_a_pruned_store_still_reports_the_park_the_failure_record_holds(tmp_path: Path) -> None:
+    """The fallback this PR exists for, pinned by a test.
+
+    A workspace cleanup can remove the marker file while the failure
+    records survive. Without the fallback the reader would answer "not
+    parked" for a session that never resumed. Driven through the real
+    producer rather than a hand-written record, so a change to the
+    record shape fails here instead of passing against a fixture that
+    only this test believes in.
+    """
+    sup = SpawnSupervisor(workdir=tmp_path)
+    budget = _budget(max_respawns=1)
+    assert sup.record_spawn_failure("batch:T-pruned", RuntimeError("boom"), budget=budget) is True
+    assert sup.record_spawn_failure("batch:T-pruned", RuntimeError("boom")) is False
+
+    _store(tmp_path).unlink()
+
+    result = load_parked_sessions(tmp_path)
+    assert result.available
+    assert result.session_ids == frozenset({"batch:T-pruned"})
+
+
+def test_a_cleared_park_is_not_resurrected_by_its_failure_record(tmp_path: Path) -> None:
+    """Clearing has to survive the fallback.
+
+    A ``respawn_exhausted`` record says a session was once exhausted --
+    history, not present state. Read as a second source and unioned in,
+    it puts back every id an operator resume or ``clear_parked`` has
+    removed, and the marker it was cleared from keeps saying nothing
+    about it. The fallback therefore runs only when the marker gave no
+    answer at all.
+    """
+    sup = SpawnSupervisor(workdir=tmp_path)
+    budget = _budget(max_respawns=1)
+    assert sup.record_spawn_failure("batch:T-cleared", RuntimeError("boom"), budget=budget) is True
+    assert sup.record_spawn_failure("batch:T-cleared", RuntimeError("boom")) is False
+    assert load_parked_sessions(tmp_path).session_ids == frozenset({"batch:T-cleared"})
+
+    assert sup.clear_parked("batch:T-cleared") is True
+
+    result = load_parked_sessions(tmp_path)
+    assert result.available
+    assert result.session_ids == frozenset()

